### PR TITLE
[Modal]Fix try-catch syntax error

### LINKED ISSUE #4552: Try-catch Syntax error in Modal
## Issue summary

After 7.0.0 update, try-catch syntax errors happens in any Modal component usages in some Android 

## Expected behavior

No syntax error

## Actual behavior

```
SyntaxError: Unexpected token {
  at ? (./node_modules/@shopify/polaris/build/esm/components/Modal/Modal.js:69:9)
```

```
    const iframe = evt.target;
    if (iframe && iframe.contentWindow) {
      try {
        setIframeHeight(iframe.contentWindow.document.body.scrollHeight);
      } catch {
        setIframeHeight(DEFAULT_IFRAME_CONTENT_HEIGHT);
      }
    }
    if (onIFrameLoad != null) {

```

## Steps to reproduce the problem

Any Modal component use.

Nextjs 

## Reduced test case
-

## Specifications

- Are you using the React components? Y
- Polaris version number: 7.0.0
- Browser: Chrome Mobile Webview (62.0.3202)
- Device: Vivo 1716
- Operating System: Version:8.1.0

### PR DESCRIPTION
**Pull request by youminkim**
### WHY are these changes introduced?

Fixes #4552  

### WHAT is this pull request doing?

Added identifier in catch to avoid syntax error

### CODE REVIEW

**Comment by ghost** [comment]
👋 Thanks for opening your first pull request. A contributor should give feedback soon. If you haven’t already, please check out the [contributing guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md).

**Comment by alex-page** [comment]
Hi @youminkim I don't think this will resolve the issue you are experiencing. This PR adds the error argument to the catch and it isn't used. As long as the `} catch {` code block gets triggered (which is intentional) it looks like your tool will call it as an error.

I am going to close this PR but feel free to re-open if I misunderstood.

**Comment by alex-page** [comment]
I have created a new PR here https://github.com/Shopify/polaris-react/pull/4625

**Comment by youminkim** [comment]
Hi @alex-page My fix was to include error variable in try catch syntax. I suspected some browsers has issue without error variable 
https://2ality.com/2017/08/optional-catch-binding.html

**Comment by alex-page** [comment]
@youminkim we are going to use your solution. Thanks for the context. Are you able to add a changlog entry?

**Review by alex-page — APPROVED** [review]

File: src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ export const Modal: React.FunctionComponent<ModalProps> & {
       if (iframe && iframe.contentWindow) {
         try {
           setIframeHeight(iframe.contentWindow.document.body.scrollHeight);
-        } catch {
+        } catch (err) {
```
Comment (kyledurand) [inline src/components/Modal/Modal.tsx]: We typically use something like:

```suggestion
        } catch (_) {
```

or 
```suggestion
        } catch (_error) {
```

To show that we're not using a var that needs to be declared.

File: UNRELEASED.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Bug fixes
 
+- Fixed try-catch syntax error in Modal ([#4553](https://github.com/Shopify/polaris-react/pull/4553))
```
Comment (kyledurand) [inline UNRELEASED.md]:
```suggestion
- Fixed try-catch syntax error in `Modal` ([#4553](https://github.com/Shopify/polaris-react/pull/4553))
```

**Review by kyledurand — APPROVED** [review]
Just one minor comment but otherwise this looks good to me. Thanks @youminkim!

**Pull request merged**